apps/sms/views/inbox/js/inbox.js

```diff
@@ -806,29 +708,16 @@ var InboxView = {
     Threads.delete(threadId);
 
     // Cleanup the DOM
-    this.removeConversationDOM(threadId);
+    var index = this.listData.findIndex((data) => data.id === threadId);
+    if (index > -1) {
+      this.listData.splice(index, 1);
+      this.list.setModel(this.listData);
```

Comment:
__question:__ also it's still not clear to me, how we're going to handle language change scenario - should we just iterate through entire datasource and re-localize everything and then re-set list model?